contracts/BufferLib.sol

```diff
@@ -12,6 +12,12 @@ library BufferLib {
     uint64 cursor;
   }
 
+  // Ensures we access an existing index in an array
+  modifier notOutOfBounds(uint256 index, uint256 length) {
```

Comment:
I'd say that for the scope of `BufferLib`, the `index` argument could use `uint32`.

Reply:
the problem is buffer.data.length is uint256, and cursor uint64. Therefore I would have to do an implicit conversion.